File: TransCoder_tokenized_test_set_functions/python/LEXICOGRAPHICALLY_SMALLEST_STRING_WHOSE_HAMMING_DISTANCE_GIVEN_STRING_EXACTLY_K.py
def findString ( str, n, k ) :
    if ( k == 0 ) :
        print ( str )
        return
    str2 = str
    p = 0
    for i in range ( 0, n, 1 ) :
        if ( str2 [ i ] != 'a' ) :
            str2 = str2 . replace ( str2 [ i ], 'a' )
            p += 1
            if ( p == k ) :
                break
    if ( p < k ) :
        i = n - 1
        while ( i >= 0 ) :
            if ( str [ i ] == 'a' ) :
                str2 = str2 . replace ( str2 [ i ], 'b' )
                p += 1
            if ( p == k ) :
                break
            i -= 1
    print ( str2 )

File: TransCoder_tokenized_test_set_functions/java/LEXICOGRAPHICALLY_SMALLEST_STRING_WHOSE_HAMMING_DISTANCE_GIVEN_STRING_EXACTLY_K.java
static void findString ( String str, int n, int k ) {
  if ( k == 0 ) {
    System . out . println ( str );
    ;
    return;
  }
  String str2 = str;
  int p = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( str2 . charAt ( i ) != 'a' ) {
      str2 = str2 . substring ( 0, i ) + 'a' + str2 . substring ( i + 1 );
      p ++;
      if ( p == k ) break;
    }
  }
  if ( p < k ) {
    for ( int i = n - 1;
    i >= 0;
    i -- ) if ( str . charAt ( i ) == 'a' ) {
      str2 = str2 . substring ( 0, i ) + 'b' + str2 . substring ( i + 1 );
      p ++;
      if ( p == k ) break;
    }
  }
  System . out . println ( str2 );
}